### PR TITLE
Proposal remove npm dependency

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

-  This proposal is based on the use of:
    - [x] "wikimedia/composer-merge-plugin".
    - [x] "https://asset-packagist.org"
    - [x] "oomphinc/composer-installers-extender": "@dev"

Config composer.json basic:
```yaml
"require": {
    "php": "^7.4|^8.0",
    "wikimedia/composer-merge-plugin": "*"
},
"extra": {
    "branch-alias": {
        "dev-master": "1.0.x-dev"
    },
    "installer-types": [
        "npm-asset"
    ],
    "installer-paths": {
        "./node_modules/{$name}": [
            "type:npm-asset"
        ]
    },
    "merge-plugin": {
        "include": [
            "composer.assets.json"
        ]
    }
},
"repositories": [
    {
        "type": "composer",
        "url": "https://asset-packagist.org"
    }
]
```
composer.assets.json:
```yaml
"require": {
    "npm-asset/vizuaalog--bulmajs": "^0.11.0",
    "npm-asset/bulma": "^0.9.1",
    "npm-asset/bulma-helpers": "^0.3.12",
    "oomphinc/composer-installers-extender": "@dev"
}
```

As a result we get all the assets in ./node_modules in the same way as if we had used npm.  

Note:

We must wait for the compatibility as composerV2 that is already made.

https://github.com/wikimedia/composer-merge-plugin/pull/189

### CODE REVIEW

**Comment by samdark** [comment]
Foxy could be removed from yiisoft/app if this is merged.

File: composer.json [hunk 1]
```diff
@@ -20,16 +20,16 @@
     "prefer-stable": true,
     "require": {
         "php": "^7.4|^8.0",
-        "foxy/foxy": "^1.1@dev",
+        "wikimedia/composer-merge-plugin": "dev-feature/composer-v2",
```
Comment (samdark) [inline composer.json:23]: Is that needed? Yii 2 worked w/o it relying on the same asset packagist well.

Comment (terabytesoftw) [reply composer.json]: done.

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,10 @@
     "prefer-stable": true,
     "require": {
         "php": "^7.4|^8.0",
-        "foxy/foxy": "^1.1@dev",
+        "npm-asset/vizuaalog--bulmajs": "^0.11.0",
+        "npm-asset/bulma": "^0.9.1",
+        "npm-asset/bulma-helpers": "^0.3.12",
+        "oomphinc/composer-installers-extender": "@dev",
```
Comment (samdark) [inline composer.json:26]: There's `2.0.0` available.

Comment (terabytesoftw) [reply composer.json:26]: yes work fine.

Comment (samdark) [reply composer.json:26]: I mean dependencies are better to be tied to stable versions in case these are available.

Comment (terabytesoftw) [reply composer.json]: done.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**